lib/sudo-exec/src/monitor.rs

```diff
@@ -74,38 +78,35 @@ impl MonitorRelay {
     pub(super) fn run(mut self) -> io::Result<std::convert::Infallible> {
         loop {
             // First we check if the command is finished
-            self.wait_command()?;
-
-            // Then we check any pending signals that we received. Based on `mon_signal_cb`
-            for info in self.signals.wait() {
-                self.relay_signal(info);
-            }
-        }
-    }
+            if let Some(status) = self.wait_command()? {
+                ExitReason::from_status(status).send(&self.tx)?;
 
-    fn wait_command(&mut self) -> io::Result<()> {
-        if let Some(status) = self.command.try_wait()? {
-            ExitReason::from_status(status).send(&self.tx)?;
+                // Given that we overwrote the default handlers for all the signals, we must
+                // emulate them to handle the signal we just sent correctly.
+                drop(self.signal_handlers);
 
-            // Given that we overwrote the default handlers for all the signals, we musti
-            // emulate them to handle the signal we just sent correctly.
-            for info in self.signals.pending() {
-                emulate_default_handler(info.signal)?;
+                exit(0);
             }
 
-            exit(0);
+            // Then we check any pending signals that we received. Based on `mon_signal_cb`
+            if let Ok(infos) = self.signal_handlers.poll() {
```

Comment:
this `poll` is a blocking operation, right? shouldn't the preceding `wait_command` / `Command::try_wait` be included in the poll set? otherwise this could stay in blocking mode (maybe forever?) meanwhile `wait_command` occurs

Reply:
right, `ErrorKind::Interrupt` is a thing. I think that detail should be documented in a comment.

what about error handling then. right now all errors are ignored but maybe only `ErrorKind::Interrupt` should be ignored and all types of errors should be bubbled up?

Reply:
> That's the plan but I didn't want to do that on this PR to avoid having too many changes at once.

would be good to add a TODO comment about this (including the try_wait in the poll set)